### DISCUSSION TITLE
Clamp in size meaning

### DISCUSSION THREAD

**Discussion by zeoint** [mienaiyami/yomikiru#158] [Q&A] [answered]
What does enabling clamp under size in book reader do?
I do not see any definition when I hover on it and also did not find any explanation to it in app settings.
Can you please explain what it does?

![1](https://github.com/mienaiyami/yomikiru/assets/127277950/fbeb1b3e-2e1f-4246-9060-b0cdad8247e4)

https://github.com/mienaiyami/yomikiru/assets/127277950/5f5f751d-fb3a-4478-913d-2183cdfc692c

**Comment by mienaiyami** [comment, marked as answer]
Limits width of image to screen and it enables you to view images at their full resolution.
You wont see it if the image's resolution is smaller than the screen's.

**Reply by zeoint** [reply]
Isn't it the same as zoom then?

**Reply by mienaiyami** [reply]
Yes, with this you can go above 100% size, if the image allows it (when image width resolution is higher than screen width.

**Reply by zeoint** [reply]
This I did not understand, can you please rephrase?

If you go above 100% size, then the image would not be visible right?

**Reply by mienaiyami** [reply]
In early versions you could go only up to 100% of screen width, I dont really remember well bit this option was needed. If it is enabled you cant go above 100% size with shortcuts and such. Also Let you want to see image at its real resolution 1:1, if the width is limited to 100%, image will only grow upto 100% of screen but it is not 1:1, for the real 1:1 you will need to disable clamping. 

There was a important reason but I cant seem to remember it.

**Reply by mienaiyami** [reply]
Maybe it was in case of 2page spanning images in manga, that a option to remove 100% width limit was needed.

**Reply by mienaiyami** [reply]
#3 , this issue was closed on the day I added clamp option.

**Reply by zeoint** [reply]
Cool, now I get it. Thanks.
